lib/topological_inventory/orchestrator/health_check.rb

```diff
@@ -0,0 +1,62 @@
+module TopologicalInventory
+  module Orchestrator
+    class HealthCheck
+      include Logging
+
+      attr_accessor :sources, :topology, :k8s
+
+      def initialize(sources, topology, k8s)
+        @sources = sources
+        @topology = topology
+        @k8s = k8s
+      end
+
+      def run
+        loop do
+          checks
+
+          sleep configured_seconds
+        end
+      end
+
+      def checks
+        errors = []
+
+        errors << check_url(topology)
+        errors << check_url(sources)
+        errors << check_k8s(k8s)
+
+        if errors.any?
+          FileUtils.rm_f("/tmp/healthy")
+        else
+          FileUtils.touch("/tmp/healthy")
+        end
+      end
+
+      private
+
+      def check_url(uri)
+        logger.info("[HealthCheck] Checking URL: #{uri}")
+
+        resp = Net::HTTP.get_response(uri.host, "/health", uri.port)
+        raise if resp.code.to_i != 200
+      rescue
+        logger.warn("[HealthCheck] Failed URL: #{uri}")
+        :error
+      end
+
+      def check_k8s(k8s)
+        logger.info("[HealthCheck] Checking k8s: #{k8s.send(:connection).api_endpoint}")
+
+        raise if k8s.check_api_status.nil?
+      rescue
+        logger.warn("[HealthCheck] k8s failing")
+        :error
+      end
+
+      def configured_seconds
+        @configured_seconds ||= ENV['HEALTH_CHECK_INTERVAL'] ? ENV['HEALTH_CHECK_INTERVAL'].to_i : 60
```

Comment:
@lindgrenj6 can you make it more visible? At least add comment to `/bin/topological_inventory-orchestrator`

Reply:
I mean the ENV variable

Reply:
Added to `bin/topological_inventory-orchestrator`